swarm/api/http/server.go

```diff
@@ -890,8 +913,17 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	w.Header().Set("Content-Type", contentType)
-	http.ServeContent(w, r, "", time.Now(), newBufferedReadSeeker(reader, getFileBufferSize))
+	if contentType != "" {
+		w.Header().Set("Content-Type", contentType)
+	}
+
+	fileName := uri.Addr
+	if found := path.Base(uri.Path); found != "" && found != "." && found != "/" {
+		fileName = found
+	}
+	w.Header().Set("Content-Disposition", fmt.Sprintf("inline; filename=\"%s\"", fileName))
```

Comment:
@lmars I think this should be:
if contentType == 'application/*' - return attachment
else return inline.
WDYT?

Reply:
Worry about "application/pdf" and "application/rss+xml", I'm personally fine if .pdf opening in browser.

Reply:
i've seen the rss issue too but i think its a non-issue as its a machine readable format and is not supposed to be read by the user as-is (AFAIK)

Reply:
ok - keep it as it is for now (`inline` it)